### PR TITLE
Improve collection of Assisted Installer related resources

### PR DESCRIPTION
**Pull request by omertuc**
**Related Issue:**  stolostron/backlog#<ISSUE_NUMBER>

None

**Description of Changes:**

Stop collecting `baremetalhost.metal3.io` and `clusterImageSet.hive.openshift.io` under the `# Inspect Assisted-installer CRs` comment because they're already collected in a difference place in the script

Start collecting missing:
    * `agent.agent-install.openshift.io`
    * `agentclassification.agent-install.openshift.io`
    * `agentserviceconfig.agent-install.openshift.io`
    * `hypershiftagentserviceconfig.agent-install.openshift.io`
    * `nmstateconfig.agent-install.openshift.io`

Also added collection of CRDs to aid automated must-gather reading tools (and it's nice to have in gathers in general)

**What resource is being added**: 

See description

**Is this a Hub or Managed cluster change?:**

Hub

**Notes:**

None

### CODE REVIEW

File: collection-scripts/gather [hunk 1]
```diff
@@ -217,10 +219,13 @@ case "$CLUSTER" in
     oc adm inspect ansiblejobs.tower.ansible.com --all-namespaces  --dest-dir=must-gather
 
     # Inspect Assisted-installer CRs
-    oc adm inspect infraenv.agent-install.openshift.io --all-namespaces --dest-dir=must-gather
-    oc adm inspect clusterImageSet.hive.openshift.io --all-namespaces --dest-dir=must-gather
```
Comment (eranco74) [inline collection-scripts/gather:221]: It's not an assisted-installer CR but why did you remove it? is it gathered elsewhere?
Same q for the BMH.

Comment (omertuc) [reply collection-scripts/gather:221]: >Stop collecting baremetalhost.metal3.io and clusterImageSet.hive.openshift.io under the # Inspect Assisted-installer CRs comment because they're already collected in a difference place in the script

**Comment by omertuc** [comment]
Seems all tests are passing, can we merge this?

**Review by jlpadilla — APPROVED** [review]
/lgtm
